cypress/e2e/login.cy.js

```diff
@@ -1,11 +1,59 @@
-/// <reference types="cypress" />
+describe("Login page", () => {
+  it("should check visiting login page", () => {
+    cy.visit("https://the-internet.herokuapp.com/login");
 
-describe('Login page', () => {
-  beforeEach(() => {
-    
+    cy.contains("Login Page");
+
+    cy.url().should("include", "login");
   });
 
-  it('', () => {
-    
+  it("should login user with valid creds", () => {
+    cy.visit("https://the-internet.herokuapp.com/login");
+
+    cy.get('[name="username"]').type("tomsmith");
+    cy.get('[name="password"]').type("SuperSecretPassword!");
+
+    cy.get(".fa").click();
+
+    cy.get(".flash").should("contain", "You logged into a secure area!");
+
+    cy.url().should("include", "secure");
   });
 });
+
+it("Unable to login with invalid login", () => {
+  cy.visit("https://the-internet.herokuapp.com/login");
+
+  cy.get('[name="username"]').type("emmasmith");
+  cy.get('[name="password"]').type("BadSecretPassword!");
+
+  cy.get(".fa").click();
+
+  cy.get(".flash").should("contain", "Your username is invalid!");
+});
+
+it("Unable to login with invalid password", () => {
```

Comment:
```suggestion
it("should not allow to log in with invalid password", () => {
```

you test case name should continue the describe name